key4hep/BitFieldDecoder/BitFieldDecoder/decoder.py

```
@@ -0,0 +1,131 @@
+                Example: "field1:8,field2:-16" (8-bit unsigned and 16-bit signed).
+        """
+        self.fields = {}
+        self.current_value = 0
+        offset = 0
+
+        for field_desc in description.split(","):
+            try:
+                name, width = field_desc.split(":")
+                width = int(width)
+            except ValueError:
+                raise ValueError(f"Invalid field description: {field_desc}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)